File: src/test/java/de/florianschmitt/rest/base/DBOperationLookup.java
package de.florianschmitt.rest.base;

import org.dbunit.operation.CompositeOperation;

import com.github.springtestdbunit.annotation.DatabaseOperation;
import com.github.springtestdbunit.operation.DefaultDatabaseOperationLookup;

public class DBOperationLookup extends DefaultDatabaseOperationLookup {

    @Override
    public org.dbunit.operation.DatabaseOperation get(DatabaseOperation operation) {
        org.dbunit.operation.DatabaseOperation superResult = super.get(operation);

        org.dbunit.operation.DatabaseOperation result = new CompositeOperation(
                new org.dbunit.operation.DatabaseOperation[]{ //
                        ChangeReferencialIntegrityOperation.disable(), //
                        superResult, //
                        ChangeReferencialIntegrityOperation.enable()// , //
                });

        return result;
    }
}

File: src/test/kotlin/de/florianschmitt/base/DBOperationLookup.kt
package de.florianschmitt.base

import org.dbunit.operation.CompositeOperation

import com.github.springtestdbunit.annotation.DatabaseOperation
import com.github.springtestdbunit.operation.DefaultDatabaseOperationLookup

class DBOperationLookup : DefaultDatabaseOperationLookup() {

    override fun get(operation: DatabaseOperation): org.dbunit.operation.DatabaseOperation {
        val superResult = super.get(operation)

        return CompositeOperation(
                arrayOf(//
                        ChangeReferencialIntegrityOperation.disable(), //
                        superResult, //
                        ChangeReferencialIntegrityOperation.enable()// , //
                ))
    }
}